packages/ui-collectionview/index.ios.ts

```diff
@@ -590,46 +591,46 @@ export class CollectionView extends CollectionViewBase {
+                        const identifier = this._dataSource.itemIdentifierForIndexPath(indexPath) || getUUID();
+                        // console.log(' splice, update identifier:', identifier)
+                        addIdentifiers.push(identifier);
+                    }
+                    this.modifyDataSourceSnapshot(ChangeType.Update, addIdentifiers, sectionIdentifier);
+                } else {
+                    if (event.removed && event.removed.length > 0) {
+                        const deleteIdentifiers = [];
+                        for (let index = 0; index < event.removed.length; index++) {
+                            const indexPath = NSIndexPath.indexPathForItemInSection(event.index + index, sectionIdentifier);
+                            const identifier = this._dataSource.itemIdentifierForIndexPath(indexPath);
+                            // console.log(' splice, remove identifier:', identifier)
+                            deleteIdentifiers.push(identifier);
                         }
-                        if (event.addedCount > 0) {
-                            const identifiers = [];
-                            for (let index = 0; index < event.addedCount; index++) {
-                                const indexPath = NSIndexPath.indexPathForItemInSection(event.index + index, sectionIdentifier);
-                                const identifier = this._dataSource.itemIdentifierForIndexPath(indexPath) || getUUID();
-                                // console.log(' splice, add identifier:', identifier)
-                                identifiers.push(identifier);
-                            }
-                            this.modifyDataSourceSnapshot(ChangeType.Add, identifiers, sectionIdentifier);
+                        this.unbindUnusedCells(event.removed);
+
+                        this.modifyDataSourceSnapshot(ChangeType.Delete, deleteIdentifiers, sectionIdentifier, false, false);
+                    }
+                    if (event.addedCount > 0) {
+                        const addIdentifiers = [];
+                        for (let index = 0; index < event.addedCount; index++) {
+                            const indexPath = NSIndexPath.indexPathForItemInSection(event.index + index, sectionIdentifier);
+                            const identifier = this._dataSource.itemIdentifierForIndexPath(indexPath) || getUUID();
+                            // console.log(' splice, add identifier:', identifier)
+                            addIdentifiers.push(identifier);
                         }
+                        this.modifyDataSourceSnapshot(ChangeType.Add, addIdentifiers, sectionIdentifier, false, false);
                     }
-                    // view.collectionViewLayout.invalidateLayout();
-                return;
+                }
+                break;
             }
         }
-        this.refresh();
+        view.contentOffset = contentOffset;
+        //this.refresh();
```

Comment:
curious on this here - simply setting contentOffset back vs. refresh? Any adverse affects seen?

Reply:
this.refresh() would be needed for some scenarios, ngZone comes to mind.